### ISSUE TITLE
I got TS error, but nuxt is not breaking the build

### ISSUE THREAD

**Issue by buglavecz** [nuxt/typescript#593]
**Describe the bug**
We want use nuxt 2 with typescript support, but I don't understand why don't break the build, if the TS code is bad?

**Screenshots**
![image](https://user-images.githubusercontent.com/43108336/234537315-8125aab7-a003-4c74-bdbe-36b878b03fc7.png)

**Additional context**
I tried add the `--fail-on-error` flag to package.json, but not working.... 🤔 
```
"scripts": {
    "dev": "cross-env BUILD_VER=dev-local.0123456 COMMIT_HASH=0123456 nuxt --fail-on-error",
```

Thanks.

**Comment by buglavecz** [comment]
I seems this error come only if I use composition API, and script setup syntax.

**Comment by rchl** [comment]
If you provide a repo I can try and see if I can reproduce. But again, if Nuxt and Volar doesn't complain then it sounds like an issue with Editor integration that does the type checking.

**Comment by buglavecz** [comment]
With this syntax setup syntax...
```vue
<template>
  <b-container class="mt-3">
    <h3>Composition API</h3>
    {{ fruit }}
  </b-container>
</template>

<script lang="ts">
    import { defineComponent } from 'vue';
    export default defineComponent({
        setup () {
            interface Apple {
                id: number,
                name: string,
                weight?: number
            }

            const fruit:Apple[] = [
                {
                    id: '10',
                    name: 'jonatan'
                },
                {
                    id: 20,
                    name: 'golden'
                }
            ];
            return { fruit };
        }
    });
</script>
```
I got this TS error in terminal:
![image](https://user-images.githubusercontent.com/43108336/234817106-d8be4bcd-66bc-4585-bf2e-2406e3d536ba.png)

But, If I use <script setup> syntax:
```vue
<template>
  <b-container class="mt-3">
    <h3>Composition API</h3>
    {{ fruit }}
  </b-container>
</template>

<script setup lang="ts">
    interface Apple {
        id: number,
        name: string,
        weight?: number
    }

    const fruit:Apple[] = [
        {
            id: '10',
            name: 'jonatan'
        },
        {
            id: 20,
            name: 'golden'
        }
    ];
</script>
```
I got this result: 
![image](https://user-images.githubusercontent.com/43108336/234817550-f445feb6-196d-4886-a997-bc692c692108.png)

this is very simple example, but I cat set a repo.

**Comment by rchl** [comment]
it would help if you can setup a repo yes.

**Comment by buglavecz** [comment]
Repo: https://github.com/buglavecz/nuxt2-bootstrap-vue-ts-starter

Stackblitz: https://stackblitz.com/github/buglavecz/nuxt2-bootstrap-vue-ts-starter

thanks, for help.

**Comment by buglavecz** [comment]
Updated the repo with `@nuxtjs/composition-api`..

**Comment by buglavecz** [comment]
![image](https://user-images.githubusercontent.com/43108336/234851867-0963d7f1-4cb8-4497-832f-4de942b30439.png)

![image](https://user-images.githubusercontent.com/43108336/234851986-87db9230-d88e-40bc-aec4-f83005c5f2d4.png)

**Comment by rchl** [comment]
Unfortunately `script setup` is just not supported in the `fork-ts-checker-webpack-plugin` webpack plugin. See authors comment about whole vue support: https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/issues/668#issuecomment-1455196701

I suppose the recommendation for now would be to just disable `typecheck` in this module and use Volar's `vue-tsc` on the command line instead.
